test/e2e/framework/azure/eventhub.go

```diff
@@ -0,0 +1,105 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"time"
+
+	eventhubs "github.com/Azure/azure-event-hubs-go/v3"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/eventhub/armeventhub"
+
+	"github.com/Azure/go-autorest/autorest/to"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+)
+
+func CreateEventHubNamespaceOnly(ctx context.Context, subscriptionID, name, region, rg string) *eventhubs.Hub {
+	return CreateEventHubCommon(ctx, subscriptionID, name, region, rg, true)
+}
+
+func CreateEventHubComponents(ctx context.Context, subscriptionID, name, region, rg string) *eventhubs.Hub {
+	return CreateEventHubCommon(ctx, subscriptionID, name, region, rg, false)
+}
+
+func CreateEventHubCommon(ctx context.Context, subscriptionID, name, region, rg string, omitHub bool) *eventhubs.Hub {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		framework.FailfWithOffset(1, "Unable to authenticate: %s", err)
+	}
+
+	nsClient := armeventhub.NewNamespacesClient(subscriptionID, cred, nil)
+	ehClient := armeventhub.NewEventHubsClient(subscriptionID, cred, nil)
+
+	// create the eventhubs namespace
+	nsResp, err := nsClient.BeginCreateOrUpdate(ctx, rg, name, armeventhub.EHNamespace{
+		TrackedResource: armeventhub.TrackedResource{
+			Location: &region,
+			Tags:     map[string]*string{E2EInstanceTagKey: to.StringPtr(name)},
+		},
+		Identity: &armeventhub.Identity{
+			Type: armeventhub.ManagedServiceIdentityTypeNone.ToPtr(),
+		},
+		SKU: &armeventhub.SKU{
+			Name:     armeventhub.SKUNameBasic.ToPtr(),
+			Capacity: to.Int32Ptr(1),
+			Tier:     armeventhub.SKUTierBasic.ToPtr(),
+		},
+	}, nil)
+
+	if err != nil {
+		framework.FailfWithOffset(1, "Unable to create eventhub namespace: %s", err)
+	}
+
+	_, err = nsResp.PollUntilDone(ctx, time.Second*30)
+	if err != nil {
+		framework.FailfWithOffset(1, "Unable to create eventhub namespace: %s", err)
+	}
+
+	if !omitHub {
+		ehResp, err := ehClient.CreateOrUpdate(ctx, rg, name, name, armeventhub.Eventhub{
+			Properties: &armeventhub.EventhubProperties{
+				MessageRetentionInDays: to.Int64Ptr(1),
+				PartitionCount:         to.Int64Ptr(2),
+			},
+		}, nil)
+
+		if err != nil {
+			framework.FailfWithOffset(1, "Unable to create eventhub: %s", err)
+			return nil
+		}
+
+		keys, err := nsClient.ListKeys(ctx, rg, *ehResp.Name, "RootManageSharedAccessKey", nil)
+		if err != nil {
+			framework.FailfWithOffset(1, "Unable to obtain the connection string: %s", err)
+			return nil
+		}
```

Comment:
I was running the tests locally with the minimum set of permissions possible, and accidentally stumbled upon an error here because I hadn't assigned `Microsoft.EventHub/namespaces/authorizationRules/listKeys/action`.

Long story short: I was wondering why we do all this trickery to read a "root" key and assemble a connection string, when it is absolutely possible to use the exact same credentials as above to obtain an Event Hubs client? (we do it [in the reconciler](https://github.com/triggermesh/triggermesh/blob/33b9cc4196de99621a93966cf14c4eabffee6e3c/pkg/sources/client/azure/eventgrid/client.go#L73-L74), as an example)

Reply:
Think this was a holdover from an earlier version (before realizing I needed to pass `/v3` for the eventhub sdk) which required the key in addition to the connection string. Will clean this up.